### ISSUE TITLE
deleted connections should always show as deleted to the user immediately

### ISSUE THREAD

**Issue by jrhizor** [airbytehq/airbyte#10666]
For https://github.com/airbytehq/oncall/issues/155, we see a case where the connection status was not updated. While the actual API failed, it left the user's connection in a state where it appeared active.

We can accomplish this a number of ways: rollbacks to have a transactional relationship, etc.

**Comment by davinchia** [comment]
This happens here today:
https://github.com/airbytehq/airbyte/blob/master/airbyte-server/src/main/java/io/airbyte/server/handlers/ConnectionsHandler.java#L322-L331

The operation is mostly likely interrupted because the server dies before it can execute the db delete connection.

Easiest way to do this is to moving this into a temporal activity and have it rollback everything if any one of the operations fail.

**Comment by pmossman** [comment]
Grooming Notes:
- What does a rollback look like? If database persistence fails within the activity, and the activity fails all of its configured retries, the connection manager workflow should be continued-as-new
- We should try to figure out how to reliably recreate the partial delete (seems like we don't know how we actually got into this problematic state).
- We can pull some extra info for this from Stackdriver, we saw multiple calls to the deletion endpoint, the first of which had a 500 error, but then subsequent calls failed (probably because the workflow was already in a terminated state).

**Comment by davinchia** [comment]
looks like https://github.com/airbytehq/oncall/issues/161 might be related

**Comment by benmoriceau** [comment]
https://cloud.airbyte.io/workspaces/9b088314-e4db-4fb4-84a0-24ffe8ab9e0a/connections/70ed1bc1-2a87-4c4c-aa9a-80c7e369afd1 could be another instance of it. It "could" because it is possible that it comes from the the fact that the user disable the connection after deleting it. In such case we won't see any event in temporal.

**Comment by pmossman** [comment]
Pulling some logs for the connection that Davin linked, and it looks like there was an API call to `connections/delete` with the wrong request body?

![image](https://user-images.githubusercontent.com/7798112/156840748-04664900-08f7-40dc-9cf2-743a797c251f.png)

Note the first call is posting a body with `configId` and `configTypes` instead of `connectionId`. It looks like the only request body we expect with those two fields is the `JobListRequestBody`: https://github.com/airbytehq/airbyte/blob/master/airbyte-api/src/main/openapi/config.yaml#L3161-L3165

I wonder if the bad request body is somehow causing the inconsistent state, will investigate this further.

**Comment by pmossman** [comment]
@cgardens / @benmoriceau I suspect we have a bug in our RequestLogger but I don't currently understand the RequestLogger class enough to confirm. Based the screenshot above, it sure looks like we're generating log lines with incorrect request bodies.

In the RequestLogger, it looks like this is the only spot in code where the requestBody is updated: https://github.com/airbytehq/airbyte/blob/master/airbyte-server/src/main/java/io/airbyte/server/RequestLogger.java#L62

In the other `filter` [method](https://github.com/airbytehq/airbyte/blob/master/airbyte-server/src/main/java/io/airbyte/server/RequestLogger.java#L67), we only ever read the `requestBody` instance variable, we never set it based on the `requestContext` arg. Is that a problem?

**Comment by benmoriceau** [comment]
@pmossman I didn't have time to look at it a lot today but I think that it is a similar problem that has been encounter in https://github.com/airbytehq/oncall/issues/142 as well. I am not sure that I totally understand if this is linked. But it looks similar, because of the failure in the requestLogger because of a formatting issue.

**Comment by pmossman** [comment]
I'm reading more into Jersey logging and I think it should be safe to assume that the first filter method will always be invoked before the second, in which case the requestBody instance variable should be correct. So if there is a bug, it probably isn't obvious

**Comment by pmossman** [comment]
Looking into the recent instance of this issue, https://github.com/airbytehq/oncall/issues/177

Airbyte UI Connection Link: https://cloud.airbyte.io/workspaces/27bf556f-8f02-4fa4-be01-37561aebf593/connections/28f0b39f-bef7-4d58-9ecd-9224d52ef6b2/status

Temporal UI for workflow with two 'DeleteConnection' activity tasks: http://tui.cloud.airbyte.io/namespaces/default/workflows/connection_manager_28f0b39f-bef7-4d58-9ecd-9224d52ef6b2/d6f08f57-c03d-42cc-b51d-b62a63488628/history?format=compact&eventId=18

Temporal Event History exported from the above link:
[double_delete_connection_event_history.json.txt](https://github.com/airbytehq/airbyte/files/8274736/double_delete_connection_event_history.json.txt)

Logs for this connection at March 9th, ~9:30am +- 1 hour: https://console.cloud.google.com/logs/query;query=textPayload:%2228f0b39f-bef7-4d58-9ecd-9224d52ef6b2%22%0AAND%20NOT%20textPayload:%22POST%20200%20%2Fapi%2Fv1%2Fjobs%2Flist%22;timeRange=2022-03-09T17:27:00.000Z%2F2022-03-09T17:27:00.000Z--PT1H;summaryFields=:false:32:beginning:false;cursorTimestamp=2022-03-09T18:01:28.724869245Z?project=prod-ab-cloud-proj

Database state of the connection as of right now:
```
Name                 |Value                                 
---------------------+--------------------------------------
id                   |28f0b39f-bef7-4d58-9ecd-9224d52ef6b2  
namespace_definition |source                                
namespace_format     |${SOURCE_NAMESPACE}                   
prefix               |                                      
source_id            |e0141aef-d5bb-45d2-9b35-1fd6acac34ea  
destination_id       |9dddba1e-8ddd-4bdc-a48c-83a3df73c7a3  
name                 |default                               
catalog              |...(too long)
status               |active                                
schedule             |{"units": 24, "timeUnit": "hours"}    
manual               |false                                 
resource_requirements|null                                  
created_at           |2022-03-09 09:27:24.444 -0800         
updated_at           |2022-03-09 09:33:34.616 -0800         
source_catalog_id    |                                      
```

Things I'm looking into:
- Why are there two DeleteConnection activities scheduled at the same time?
- Why did the DeleteConnection activities apparently succeed without setting the Connection status to 'deprecated' in the database?
- Why does the database record have an updated_at of 9:33:34? What was updated?

**Comment by pmossman** [comment]
Looked at this with Benoit and while we don't have a fix quite yet, we were able to reproduce it and have some insight into the problem. 

This seems to only happens if a user clicks 'Delete' multiple times in quick succession. The first delete request succeeds, and the database record is correctly updated to 'status=DEPRECATED'.

However, when one of the duplicate request 500s, the database record is updated back to 'status=ACTIVE'. I confirmed this by querying the record status continuously during the duplicated delete requests and saw the status change from 'ACTIVE' to 'DEPRECATED' back to 'ACTIVE'.

Now that I have a lead, I will hopefully come up with a fix tomorrow!
